fdbserver/ClusterRecovery.actor.cpp

```diff
@@ -489,6 +489,8 @@ ACTOR Future<Void> trackTlogRecovery(Reference<ClusterRecoveryData> self,
 		            configuration.expectedLogSets(self->primaryDcId.size() ? self->primaryDcId[0] : Optional<Key>()))
 		    .detail("RecoveryCount", newState.recoveryCount);
 		wait(self->cstate.write(newState, finalUpdate));
+		// Purge in memory state after durability to avoid race conditions.
```

Comment:
I agree that it is safe to purge at this point (which is what I was trying to say in my initial comment above), but this in-memory purging is not needed? I thought your intention was to make the processes, including the cluster controller process, pick up the purged state at a later stage (which happens when the serialized purged state is read by them, probably when doing a follow up recovery)?

Reply:
I missed that part. Thanks for the explanation!